lib/rule-tester/helpers.js

```diff
@@ -0,0 +1,310 @@
+				!duplicationIgnoredParameters.has(key)
+				? value
+				: void 0;
+		},
+	});
+
+	assert(
+		!seenTestCases.has(serializedTestCase),
+		"detected duplicate test case"
+	);
+	seenTestCases.add(serializedTestCase);
+}
+
+//------------------------------------------------------------------------------
+// Public Interface
+//------------------------------------------------------------------------------
+
+/**
+ * Asserts that a rule is valid.
+ * A valid rule must be an object with a `create` method.
+ * @param {Object} rule The rule to check.
+ * @param {string} ruleName The name of the rule.
+ * @returns {void}
+ * @throws {AssertionError} If the rule is not valid.
+ */
+exports.assertRule = function assertRule(rule, ruleName) {
+	assert.ok(
+		rule && typeof rule === "object" && typeof rule.create === "function",
+		`Rule ${ruleName} must be an object with a \`create\` method`
+	);
+};
+
+/**
+ * Asserts that a test scenario object is valid.
+ * A valid test scenario object must have `valid` and `invalid` properties, both of
+ * which must be arrays.
+ * @param {Object} test The test scenario object to check.
+ * @param {string} ruleName The name of the rule being tested.
+ * @returns {void}
+ * @throws {AssertionError} If the test scenario object is not valid.
+ */
+exports.assertTest = function assertTest(test, ruleName) {
+	assertObject(
+		test,
+		`Test Scenarios for rule ${ruleName} : Could not find test scenario object`
+	);
+
+	const hasValid = Array.isArray(test.valid);
+	const hasInvalid = Array.isArray(test.invalid);
+
+	assert.ok(
+		hasValid && hasInvalid,
+		`Test Scenarios for rule ${ruleName} : No valid or invalid test scenarios found`
+	);
+};
+
+exports.assertValidTestCase = function assertValidTestCase(
+	item,
+	seenTestCases
+) {
+	// must have x properties
+	assertString(item.code, "Test case must specify a string value for 'code'");
+
+	// optional properties
+	assertStringOrUndefined(
+		item.name,
+		`Optional test case property 'name' must be a string`
+	);
+	assertBoolOrUndefined(
+		item.only,
+		`Optional test case property 'only' must be a boolean`
+	);
+	assertFunctionOrUndefined(
+		item.before,
+		`Optional test case property 'before' must be a function`
+	);
+	assertFunctionOrUndefined(
+		item.after,
+		`Optional test case property 'after' must be a function`
+	);
+	assertStringOrUndefined(
+		item.filename,
+		`Optional test case property 'filename' must be a string`
+	);
+	assertArrayOrUndefined(
+		item.options,
+		`Optional test case property 'options' must be an array`
+	);
+
+	/*
+	 * must not have x properties
+	 * TODO: uncomment this when eslint v10
+	 * assert.ok(
+	 * 	item.errors === void 0 ||
+	 * 		item.errors === 0 ||
+	 * 		(Array.isArray(item.errors) && item.errors.length === 0),
+	 * 	"Valid test case must not have 'errors' property`"
+	 * );
+	 * assertUndefined(
+	 * 	item.output,
+	 * 	"Valid test case must not have 'output' property`"
+	 * );
+	 */
```

Comment:
These checks did not exist before, and to avoid breaking changes, they will be enabled in v10.